### PR TITLE
Improve docker image building to use a stable working directory

### PR DESCRIPTION
**Pull request by roastario**

### CODE REVIEW

File: buildSrc/src/main/groovy/net/corda/testing/ImageBuilding.groovy [hunk 1]
```diff
@@ -29,8 +29,9 @@ class ImageBuilding implements Plugin<Project> {
         }
 
         DockerCreateContainer createBuildContainer = project.tasks.create('createBuildContainer', DockerCreateContainer) {
-            File gradleDir = new File((System.getProperty("java.io.tmpdir") + File.separator + "gradle"))
-            File mavenDir = new File((System.getProperty("java.io.tmpdir") + File.separator + "maven"))
+            File baseWorkingDir = new File(System.getProperty("docker.work.dir") ? System.getProperty("docker.work.dir") : System.getProperty("java.io.tmpdir"))
```
Comment (zkiss) [inline buildSrc/src/main/groovy/net/corda/testing/ImageBuilding.groovy]: didn't you mean to do `hasPropery` in the elvis? I'm assuming groovy smart casts String to a null check, but still

File: buildSrc/src/main/groovy/net/corda/testing/KubesTest.groovy [hunk 2]
```diff
@@ -48,11 +48,10 @@ class KubesTest extends DefaultTask {
             throw new GradleException("Apache Commons compress has not be loaded, this can happen if running from within intellj - please select \"delegate to gradle\" for build and test actions")
         }
 
-        def gitSha = new BigInteger(project.hasProperty("corda_revision") ? project.property("corda_revision").toString() : "0", 36)
-        def buildId = System.hasProperty("buildId") ? System.getProperty("buildId") : "UNKNOWN_BUILD"
+        def buildId = System.hasProperty("buildId") ? System.getProperty("buildId") : (project.hasProperty("corda_revision") ? project.property("corda_revision").toString() : "0")
```
Comment (zkiss) [inline buildSrc/src/main/groovy/net/corda/testing/KubesTest.groovy]: I find chained elvises to be very hard to read in a single line. Maybe do
```
condition1 ? value1 :
  condition2 ? value2 :
  else_value3
```

File: buildSrc/src/main/groovy/net/corda/testing/ImageBuilding.groovy (hunk 1 above)
Comment (roastario) [reply buildSrc/src/main/groovy/net/corda/testing/ImageBuilding.groovy]: no unfortunately not, System.hasProperty is actually essentially an extension function from groovy. It's available on all objects - it checks if it has a property in the java sense (ie, has a getter and setter). I actually got caught out by exactly this. 

Luckily groovy allows you to evaluate all objects as true / false.

File: buildSrc/src/main/groovy/net/corda/testing/KubesTest.groovy (hunk 2 above)
Comment (roastario) [reply buildSrc/src/main/groovy/net/corda/testing/KubesTest.groovy]: :+1:

Comment (roastario) [reply buildSrc/src/main/groovy/net/corda/testing/KubesTest.groovy]: `condition1 ? value1 :
  (condition2 ? value2 : else_value3)`

I think actually looks nicer.

File: buildSrc/src/main/groovy/net/corda/testing/KubesTest.groovy [hunk 3]
```diff
@@ -48,11 +48,12 @@ class KubesTest extends DefaultTask {
             throw new GradleException("Apache Commons compress has not be loaded, this can happen if running from within intellj - please select \"delegate to gradle\" for build and test actions")
         }
 
-        def gitSha = new BigInteger(project.hasProperty("corda_revision") ? project.property("corda_revision").toString() : "0", 36)
-        def buildId = System.hasProperty("buildId") ? System.getProperty("buildId") : "UNKNOWN_BUILD"
+        def buildId = System.hasProperty("buildId") ? System.getProperty("buildId") :
```
Comment (zkiss) [inline buildSrc/src/main/groovy/net/corda/testing/KubesTest.groovy]: right, so shouldn't this be using `getProperty` instead of `hasProperty` then? :)

Comment (roastario) [reply buildSrc/src/main/groovy/net/corda/testing/KubesTest.groovy]: Yep, you see! It's so confusing...

File: testing/test_times.csv [hunk 4]
```diff
@@ -0,0 +1,3716 @@
+Test Name,Duration(ms)
```
Comment (zkiss) [inline testing/test_times.csv]: committed by mistake?

Comment (roastario) [reply testing/test_times.csv]: Yep, repushed.

**Pull request merged**